### PR TITLE
[Headers][X86] Add rounding and exception notes to conversions

### PR DESCRIPTION
**Pull request by pogo59**
Consistently describe rounding/truncating on convert intrinsics. Add notes where an out-of-range result can raise an exception.

### CODE REVIEW

**Comment by pogo59** [comment]
FYI @cflores

File: clang/lib/Headers/avxintrin.h [hunk 1]
```diff
@@ -2180,7 +2180,8 @@ _mm256_cvtepi32_pd(__m128i __a)
   return (__m256d)__builtin_convertvector((__v4si)__a, __v4df);
 }
 
-/// Converts a vector of [8 x i32] into a vector of [8 x float].
+/// Converts a vector of [8 x i32] into a vector of [8 x float]. Rounds inexact
+///    results according to the rounding control bits in the MXCSR register.
```
Comment (phoebewang) [inline clang/lib/Headers/avxintrin.h:2184]: I think we don't need to mention it. This is the default behavior if not mentioned otherwise.

Comment (pogo59) [reply clang/lib/Headers/avxintrin.h]: Discussed this internally and agreed that it is not necessary.

File: clang/lib/Headers/emmintrin.h [hunk 2]
```diff
@@ -1405,11 +1413,12 @@ static __inline__ __m128d __DEFAULT_FN_ATTRS _mm_cvtss_sd(__m128d __a,
 
 /// Converts the two double-precision floating-point elements of a
 ///    128-bit vector of [2 x double] into two signed 32-bit integer values,
-///    returned in the lower 64 bits of a 128-bit vector of [4 x i32].
+///    returned in the lower 64 bits of a 128-bit vector of [4 x i32],
+///    truncating inexact results.
```
Comment (phoebewang) [inline clang/lib/Headers/emmintrin.h:1417]: The last sentence is not necessary and has ambiguity in truncating the fractional part or truncating higher bits to 32-bit.

Comment (pogo59) [reply clang/lib/Headers/emmintrin.h:1417]: Truncation is what distinguishes this from `_mm_cvtpd_epi32` so I think it is necessary? That information used to be in the second paragraph, I shortened it and moved it up to the first paragraph, to make that difference more obvious.
Behavior on overflow was added to the second paragraph.

Do you have a different suggestion for how to organize this information?

Comment (phoebewang) [reply clang/lib/Headers/emmintrin.h:1417]: Oh, I took that simply as wrong infromation.

> Do you have a different suggestion for how to organize this information?

Not ideal, something like

```
/// Converts the two double-precision floating-point elements of a
///    128-bit vector of [2 x double] into two signed truncated
///    (rounded towards zero) 32-bit integer values,returned in the
///    most negative integer.
```

**Comment by craigflores** [comment]
Test from "craig.flores@sony.com".

**Review by phoebewang — APPROVED** [review]
Looks great, thanks!

**Review by craigflores — APPROVED** [review]

**Pull request merged**